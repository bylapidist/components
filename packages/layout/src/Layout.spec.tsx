import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Layout from './index';

test('it works', () => {
    const tree = renderer
        .create(
            <Layout
                header={<p>Header</p>}
                sidebar={<p>Sidebar</p>}
                main={<p>Main</p>}
                footer={<p>Footer</p>}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
