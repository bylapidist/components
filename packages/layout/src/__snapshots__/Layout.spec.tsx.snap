// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 45px;
  padding: 16px;
  line-height: 45px;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
  min-width: 0;
}

.c3 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  min-height: 0;
}

.c4 {
  padding: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 0;
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <p>
      Header
    </p>
  </header>
  <section
    className="c2"
  >
    <p>
      Sidebar
    </p>
  </section>
  <main
    className="c3"
  >
    <p>
      Main
    </p>
  </main>
  <footer
    className="c4"
  >
    <p>
      Footer
    </p>
  </footer>
</div>
`;
